src/utils.rs

```diff
@@ -1,40 +1,38 @@
 //! Utility functions
-use std::iter::{repeat, StepBy};
+use std::iter::StepBy;
 use std::ops::Range;
 
 #[inline(always)]
-pub fn unpack_bits<F>(buf: &mut [u8], channels: usize, bit_depth: u8, func: F)
+pub fn unpack_bits<F>(row: &[u8], buf: &mut [u8], channels: usize, bit_depth: u8, func: F)
```

Comment:
This signature should probably be `unpack_bits<F>(input: &[u8], output: &mut [u8], ...)` or similar